src/Diagnostics/Debug/StateCheck.jl

```diff
@@ -0,0 +1,593 @@
+    nt_freq::Int = nt_freq_def;
+    prec = prec_def,
+)
+    if MPI.Comm_rank(MPI.COMM_WORLD) == 0
+        println("# SC Start: creating state check callback")
+    end
+
+    ####
+    # Print fields that the call back create by this call will query
+    ####
+    nr = 0
+    for f in fields
+        print_head = true
+        Q = f[1]
+        lab = f[2]
+        V = typeof(Q).parameters[2]
+        slist = typeof(Q).parameters[2].names
+        l_s = length(slist)
+        nss = 0
+        if l_s == 0
+            println(
+                io,
+                "# SC  MPIStateArray labeled \"$lab\" has no named symbols.",
+            )
+        else
+            ns = 0
+            for s in slist
+                ns = ns + 1
+                if MPI.Comm_rank(MPI.COMM_WORLD) == 0
+                    if print_head
+                        println(
+                            io,
+                            "# SC Creating state check callback labeled \"$lab\" for symbols",
+                        )
+                    end
+                    print_head = false
+                    println(io, "# SC ", s)
+                end
+                for n in
+                    flattenednames(fieldtype(V, ns), prefix = fieldname(V, ns))
+                    nss = nss + 1
+                end
+            end
+        end
+        nr = nr + nss
+    end
+
+    ###
+    # Initialize total calls counter for the call back
+    ###
+    n_cb_calls = 0
+
+    ###
+    # Create holder for most recent stats
+    ###
+    cur_stats_dict = Dict()
+    cur_stats_flat = Array{Any}(undef, nr)
+
+    ######
+    # Create the callback
+    ######
+    cb = EveryXSimulationSteps(nt_freq) do (s = false)
+        # Track which timestep this is
+        n_cb_calls = n_cb_calls + 1
+        n_step = (n_cb_calls - 1) * nt_freq + 1
+        ns_str = @sprintf("%7.7d", n_step - 1)
+
+        ## Print header
+        nprec = min(max(1, prec), 20)
+        if MPI.Comm_rank(MPI.COMM_WORLD) == 0
+            println(
+                io,
+                "# SC +++++++++++ClimateMachine StateCheck call-back start+++++++++++++++++",
+            )
+            println(
+                io,
+                "# SC  Step  |   Label    |  Field   |                            Stats                       ",
+            )
+        end
+        h_var_fmt = "%" * sprintf1("%d", nprec + 8) * "s"
+        min_str = sprintf1(h_var_fmt, " min() ")
+        max_str = sprintf1(h_var_fmt, " max() ")
+        ave_str = sprintf1(h_var_fmt, " mean() ")
+        std_str = sprintf1(h_var_fmt, " std() ")
+        if MPI.Comm_rank(MPI.COMM_WORLD) == 0
+            println(
+                io,
+                "# SC =======|============|==========|",
+                min_str,
+                "|",
+                max_str,
+                "|",
+                ave_str,
+                "|",
+                std_str,
+                "|",
+            )
```

Comment:
```suggestion
        fmt_str = [" min() ", " max() ", " mean() ", " std() "]
        fmt_str = sprintf1.(Ref(h_var_fmt), fmt_str)
        if MPI.Comm_rank(MPI.COMM_WORLD) == 0
            print(io, "# SC =======|============|==========|")
            println(join(fmt_str, "|"))
        end
```
Might want to check that this produces the same format.